lib/connections/service.go

```diff
@@ -370,15 +382,16 @@ func (s *Service) connect() {
 	}
 }
 
-func (s *Service) shouldLimit(addr net.Addr) bool {
-	if s.cfg.Options().LimitBandwidthInLan {
+func (s *Service) shouldLimit(addr net.Addr, options config.OptionsConfiguration) bool {
+	if options.LimitBandwidthInLan {
```

Comment:
@AudriusButkevicius limiting the scope is intended. I think widing the scope again when needed is fine.